File: common/src/main/java/dev/igalaxy/nostringsattached/config/NoStringsAttachedConfig.java
package dev.igalaxy.nostringsattached.config;

import me.shedaniel.autoconfig.ConfigData;
import me.shedaniel.autoconfig.annotation.Config;
import me.shedaniel.autoconfig.annotation.ConfigEntry;

@Config(name = "nostringsattached")
public class NoStringsAttachedConfig implements ConfigData {
    @ConfigEntry.Category("Blocks")
    public boolean hideString = true;
}

File: src/main/kotlin/dev/igalaxy/nostringsattached/config/NoStringsAttachedConfig.kt
package dev.igalaxy.nostringsattached.config

import me.shedaniel.autoconfig.ConfigData
import me.shedaniel.autoconfig.annotation.Config

@Config(name = "nostringsattached")
class NoStringsAttachedConfig : ConfigData {
    var enableMod = true
    var showWhenHolding = true
    var hideInOverworld = true
    var hideInNether = true
    var hideInEnd = true
}